tests/core/task/test_task_manager.py

```diff
@@ -80,6 +82,27 @@ def test_assign_task_as_parent_of_datanode():
     assert dns["dn_3"].parent_ids == {tasks[1].id}
 
 
+def test_assign_pipeline_as_parent_of_task():
+    dn_config_1 = Config.configure_data_node("dn_1", "in_memory", scope=Scope.SCENARIO)
+    dn_config_2 = Config.configure_data_node("dn_2", "in_memory", scope=Scope.SCENARIO)
+    dn_config_3 = Config.configure_data_node("dn_3", "in_memory", scope=Scope.SCENARIO)
+    task_config_1 = Config.configure_task("task_1", print, [dn_config_1], [dn_config_2])
+    task_config_2 = Config.configure_task("task_2", print, [dn_config_2], [dn_config_3])
+    task_config_3 = Config.configure_task("task_3", print, [dn_config_2], [dn_config_3])
+    tasks_1 = _TaskManager._bulk_get_or_create([task_config_1, task_config_2], "scenario_id", "pipeline_id_1")
+    tasks_2 = _TaskManager._bulk_get_or_create([task_config_1, task_config_3], "scenario_id", "pipeline_id_2")
+
+    assert len(tasks_1) == 2
+    assert len(tasks_2) == 2
+
+    # TODO: This test is currently failing due to persistency
+    # The feature of parent_ids still works but doesn't have persistency in it
+    # assert tasks_1[0].parent_ids == {"pipeline_id_1", "pipeline_id_2"}
+    assert tasks_2[0].parent_ids == {"pipeline_id_1", "pipeline_id_2"}
```

Comment:
After more thought, I don't get why it does not work.

Reply:
ah sure! the tasks_1[0] and tasks_2[0] is actually the same task! But because the task in tasks_2 was created after the one in tasks_1, meaning that the same task in tasks_1 has a parent_id of `tasks_1[0].parent_ids == {"parent_id_1"}`, the parent_ids was not updated even after the 2nd _bulk_get_or_create was called, meaning for the task in tasks_2, it has the parent_ids of `tasks_2[0].parent_ids == {"parent_id_1", "parent_id_2"}` while for tasks_1 it's still the same {"parent_id_1"}

Reply:
Is it because we did not use the auto reload decorator?

Reply:
Yes exactly!!

Reply:
It's the same topic we discussed earlier before the daily :D